src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -366,13 +382,83 @@ public String readLastGradeTransferTime() throws PersistBlobException {
 		return lastTransferTime;
 	}
 
+	// Compute a reasonable last grade transfer time.
+	protected String ensureLastGradeTransferTime(String lastTransferTime) throws PersistBlobException, SPEEsbException {
+
+		String useTransferTime;
+
+		// if specify a particular override time use that.  This is a separate property to make it easy to
+		// override from command line.
+		useTransferTime = speproperties.getGetgrades().get("gradedaftertimeoverride");
+		if (useTransferTime != null && useTransferTime.length() > 0) {
+			return useTransferTime;
+		}
+
+		// if a time was saved use that.
+		//useTransferTime = readLastGradeTransferTime();
+		if (useTransferTime != null && useTransferTime.length() > 0) {
+			return useTransferTime;
+		}
+
+		// if a time was saved use that.
+		useTransferTime = speproperties.getGetgrades().get("gradedaftertime");
+		if (useTransferTime != null && useTransferTime.length() > 0) {
+			return useTransferTime;
+		}
+
+		M_log.error("Can not determine last graded time");
+		throw  new SPEEsbException("Can not determine last graded time");
+
+	}
+
+	// Compute a reasonable last grade transfer time.
+	protected String ensureLastGradeTransferTimeOLD(String lastTransferTime) throws PersistBlobException, SPEEsbException {
```

Comment:
What is the difference between `ensureLastGradeTransferTimeOLD` and `ensureLastGradeTransferTime`? do we need both of them?